mne/io/meas_info.py

```diff
@@ -774,6 +777,29 @@ def _check_consistency(self, prepend_error=''):
             warn('the "filename" key is misleading '
                  'and info should not have it')
 
+    def _check_nirs_ch_names(self):
+        picks = _picks_to_idx(self, 'fnirs', exclude=[], allow_empty=True)
+        ch_types = _get_channel_types(self, picks=picks)
+        for idx in picks:
+            ch_name = self['chs'][idx]['ch_name']
+            if ch_name != self['ch_names'][idx]:
+                raise RuntimeError(
+                    'Bad info: info["chs"][%d]["ch_name"] not matching '
+                    'info["ch_names"][%d]' % (idx, idx))
+            name, suffix = ch_name.split(' ')
+            if suffix != ch_types[idx]:
+                try:
+                    int(suffix)
+                except ValueError:
+                    raise RuntimeError(
+                        'Bad info: info["chs"][%d]["ch_name"] has an invalid '
+                        'suffix %s' % (idx, suffix))
+            s, d = name.split('_')
+            if not re.match('^S[0-9]+$', s) or not re.match('^D[0-9]+$', d):
+                raise RuntimeError(
+                    'Bad info: info["chs"][%d]["ch_name"] has an invalid '
+                    'name %s, should follow the S#_D# format' % (idx, name))
```

Comment:
There is already a function that check the naming of nirs channels. It goes a step further and also checks that the channels are all paired by source detectors. However, it only works now for raw and optical density data, not haemoglobin. I think it would be better to extend the existing function rather than have two different name checking functions.

The existing function takes a `freqs` argument and checks that each channel pair uses these frequencies. We could change freqs to `pair_vals` and extend the function to take either frequency pairs (`[760, 850]`) OR haemoglobin pairs (`[hbo, hbr]`)

I think this could be done in a small PR self contained PR and merged quite quickly (at least before this big PR). Johann would you like me to do it today (while your timezone is dark) or would you like to?

https://github.com/mne-tools/mne-python/blob/bbcba957bcef31af9b49262ff1435a3c45c2e1df/mne/preprocessing/nirs/nirs.py#L67

Reply:
@HanBnrd

Reply:
Sure, feel free to do this if that's okay for you, I don't have much time to work on it until the end of the week.

Reply:
Ok I will give it a go. But I hope you have time to give it a real critical review, time for you to get revenge!!

Reply:
Ok I have expanded `_check_channels_ordered()` to cover both raw, od, and hb data. It has a different API to your function, but provides the same functionality and more (also checks that the pairings are correct).

https://github.com/mne-tools/mne-python/blob/2a8d56b453ccf4e7b3b22a80bdd42d4ac604474e/mne/preprocessing/nirs/nirs.py#L80

But what might be more interesting is the tests...

https://github.com/mne-tools/mne-python/blob/2a8d56b453ccf4e7b3b22a80bdd42d4ac604474e/mne/preprocessing/nirs/tests/test_nirs.py#L206-L363

Reply:
Ok after reading the `_set_montage_fnirs` code above I think you can just use the existing `_check_channels_ordered`. But let me know if you think there is an additional use case for this.

Reply:
Hi @rob-luke, I checked to see how I could use *_check_channels_ordered()* instead of *_check_nirs_ch_names()* and it appears to be a bit difficult because it takes a *raw* object while *_set_montage()* only works with *info* objects. I think that would require an API change for *_check_channels_ordered()* or *_set_montage()* which is probably not a good idea in this PR, what do you think we should do?

Reply:
Good catch @HanBnrd, totally agree with your suggested modification and have opened it in #9280. Thanks for sticking through these gritty parts, the fNIRS support is still evolving 😉.

Reply:
Thanks @rob-luke, I'm working on the tests today and will do the switch to use the new API after that 🙂

Reply:
Cheers, I know its a bit of a grind